great_expectations/datasource/fluent/snowflake_datasource.py

```diff
@@ -26,4 +37,59 @@ class SnowflakeDatasource(SQLDatasource):
     """
 
     type: Literal["snowflake"] = "snowflake"  # type: ignore[assignment]
-    connection_string: Union[ConfigStr, SnowflakeDsn]
+    connection_string: Optional[Union[ConfigStr, SnowflakeDsn]] = None
+    # connect_args
+    account: Optional[str] = None
+    user: Optional[str] = None
+    password: Optional[str] = None
+    database: Optional[str] = None
+    schema_: Optional[str] = pydantic.Field(
+        None, alias="schema"
+    )  # schema is a reserved attr in BaseModel
+    warehouse: Optional[str] = None
+    role: Optional[str] = None
+    numpy: bool = False
```

Comment:
Reminder to add overloads to the sub file to support both methods (connection string vs individual args)